### PR TITLE
[rllib] Remove experimental eager support

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

While somewhat cool, this is causing confusion. It's probably better to punt this until we can fully support eager with a "EagerTFPolicy" base class or similar.

cc @gehring 

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

**Comment by ericl** [comment]
jenkins retest this please

File: python/ray/rllib/policy/tf_policy_template.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def build_tf_policy(name,
     This means that you can e.g., depend on any policy attributes created in
     the running of `loss_fn` in later functions such as `stats_fn`.
 
-    In eager mode (experimental), the following functions will be run
+    In eager mode (to be implemented), the following functions will be run
```
Comment (hartikainen) [inline python/ray/rllib/policy/tf_policy_template.py]: Should we just remove the eager part from the doc string to not confuse anyone?

**Review by hartikainen — APPROVED** [review]
Looks good to me

File: python/ray/rllib/policy/tf_policy_template.py (hunk 1 above)
Comment (ericl) [reply python/ray/rllib/policy/tf_policy_template.py]: I think this is probably fine, we are definitely going to add eager.

**Comment by ericl** [comment]
jenkins retest this please

**Comment by ericl** [comment]
jenkins retest this please

**Pull request merged**